### DISCUSSION TITLE
how to get a pandas dataframe of ds.data_vars?

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5309] [Q&A] [answered]
I would like to turn `ds.data_vars` into a `pandas.DataFrame` with two columns: "Data variable" and "Dimension". This helps viewing datasets with a large number of variables. I can use pandas to export to csv, to filter for certain dims etc.

```
import xarray as xr
import pandas as pd

url = "https://thredds.ucar.edu/thredds/dodsC/grib/NCEP/GFS/Global_0p25deg/Best"
ds = xr.open_dataset(url)
```

`ds.data_vars` has a nice repr layout
```
ds.data_vars
```
<img width="1250" alt="Screen Shot 2021-05-14 at 5 12 25 AM" src="https://user-images.githubusercontent.com/17162724/118249103-0b5fd280-b473-11eb-9e5a-55a9a5908294.png">

If I plug this into a pd.Dataframe I just get the variables
```
pd.DataFrame(ds.data_vars)
```
<img width="362" alt="Screen Shot 2021-05-14 at 5 15 18 AM" src="https://user-images.githubusercontent.com/17162724/118249392-6396d480-b473-11eb-9c41-53e11a58d7b0.png">

I have some workable pandas code but didn't know if the [`DataVariables` class](https://github.com/pydata/xarray/blob/master/xarray/core/dataset.py#L513) had something built in or there are better ways to do this.

```
df = pd.DataFrame(ds.data_vars, columns=["Data variable"])
d = ds.variables
l = []
for _, row in df.iterrows():
    l.append(d[row["Data variable"]].dims)
df["Dimensions"] = l
```
<img width="651" alt="Screen Shot 2021-05-14 at 5 27 09 AM" src="https://user-images.githubusercontent.com/17162724/118250992-0a2fa500-b475-11eb-959e-9ae236de0878.png">

**Comment by keewis** [comment, marked as answer]
I don't think we have anything like that, yet. For reference, it's much easier to compute all at once:
```python
In [10]: ds = xr.tutorial.open_dataset("eraint_uvz")
    ...: pd.DataFrame(
    ...:     [[name, var.dims] for name, var in ds.data_vars.items()],
    ...:     columns=["Data Variable", "Dimensions"],
    ...: )
Out[10]: 
  Data Variable                           Dimensions
0             z  (month, level, latitude, longitude)
1             u  (month, level, latitude, longitude)
2             v  (month, level, latitude, longitude)
```
